### PR TITLE
Fix usage of OMERO_SESSION_DIR

### PR DESCRIPTION
**Pull request by sbesson**
Initial implementation of OMERO_SESSION_DIR led to the creation of a omero/sessions subfolders. This behavior is counter-intuitive and does not match the CLI documentation. This commit fixes this behavior by deprecating `OMERO_SESSION_DIR` in favor of `OMERO_SESSIONDIR` which allows to set the equivalent of `$HOME/omero/sessions` by environment variable.

To test this PR, check the sessions storage mechanism under various environments:

```
# Should store under ~/omero/sessions
bin/omero login -s localhost -u user -w pass -C
bin/omero sessions file
# Should store under ~/xxx/omero/sessions
OMERO_SESSION_DIR=~/xxx bin/omero login -s localhost -u user -w pass -C
OMERO_SESSION_DIR=~/xxx bin/omero sessions file
# Should store under ~/xxx2
OMERO_SESSIONDIR=~/xxx2 bin/omero login -s localhost -u user -w pass -C
OMERO_SESSIONDIR=~/xxx2 bin/omero sessions file
# Should store under ~/xxx2
OMERO_SESSIONDIR=~/xxx2 OMERO_SESSION_DIR=~/xxx bin/omero login -s localhost -u user -w pass -C
OMERO_SESSIONDIR=~/xxx2 OMERO_SESSION_DIR=~/xxx bin/omero sessions file
```

### CODE REVIEW

**Comment by joshmoore** [comment]
So you assume that anyone who has used it to date, is actually using `$X/omero/sessions/omero/sessions` and therefore the change to just `$X/omero/sessions` should be minimally breaking?

**Comment by joshmoore** [comment]
Independent of the "breakingness", the testing of gh-3734 was much more sensible with this change in place.

**Comment by sbesson** [comment]
On the breakingness, this PR is certainly open to have this discussion before code freeze.

My major issue is that the CLI help, the [Sphinx documentation](https://www.openmicroscopy.org/site/support/omero5.1/users/cli/sessions.html?highlight=omero_session_dir#sessions-directory) and the implementation are not in-line with each other. If we estimate this code change is too breaking for 5.1.1, we can certainly close this PR and I will fix the CLI help elsewhere. /cc @ximenesuk @ehrenfeu

**Comment by ehrenfeu** [comment]
I can't say much about this right now as I didn't have a chance of testing gh-3734 yet, although it is pretty high on my priority list - hopefully I'll find the time there in the next two weeks (in my case this was linked to the HRM-OMERO-connector).

File: components/tools/OmeroPy/src/omero/plugins/sessions.py [hunk 1]
```diff
@@ -100,7 +100,7 @@
 Other commands:
 
     $ bin/omero sessions list
-    $ OMERO_SESSION_DIR=/tmp bin/omero sessions list
+    $ OMERO_SESSIONDIR=/tmp bin/omero sessions list
```
Comment (joshmoore) [inline components/tools/OmeroPy/src/omero/plugins/sessions.py]: I'd used something uniquer here, like `/tmp/$$`

Comment (sbesson) [inline components/tools/OmeroPy/src/omero/plugins/sessions.py]: Or like https://github.com/openmicroscopy/ome-documentation/pull/1194/files ? :)

**Comment by manics** [comment]
Would it be reasonable to keep the variable as `OMERO_SESSION_DIR` and just change its behaviour?

**Comment by sbesson** [comment]
@manics: this was my initial implementation (1938859). I guess the debate here is how breaking this would be. We can have a call this morning.

**Comment by manics** [comment]
If `OMERO_SESSION_DIR` is set the created session is still stored in `~/omero/sessions`:
```
$ rm -r ~/xxx/rm: cannot remove `/homes/spli/xxx/': No such file or directory

$ OMERO_SESSION_DIR=~/xxx bin/omero login -s localhost -u user-1 -w ome
<string>:144: DeprecationWarning: OMERO_SESSION_DIR is deprecated. Use OMERO_SESSIONDIR instead.
Using session 81926c0b-4eab-4eb2-b3f1-a350e6efc2de (user-1@localhost:4064). Idle timeout: 10.0 min. Current group: private-1

$ ls ~/xxx/
ls: cannot access /homes/spli/xxx/: No such file or directory

$ ls ~/omero/sessions/localhost/user-1/
81926c0b-4eab-4eb2-b3f1-a350e6efc2de
```

**Comment by sbesson** [comment]
Just realized the deprecation warning for the `--session-dir` argument needs to be wrapped in another `if base_dir` statement (else it is now printed by default on every regular `bin/omero sessions` call on Python 2.6). I will push another fix shortly but the environment logic is unmodified and can be retested following the PR description.

**Comment by manics** [comment]
good to merge

**Pull request merged**